readme.md

```diff
@@ -1,57 +1,60 @@
 # Mannish the Manly Mediator
 
-Inspired by Nicholas Zakas's talk on "[Creating A Scalable JavaScript Application Architecture](http://youtu.be/b5pFv9NB9fs)", this is a mediator to let modules communicate without tight coupling - for whatever "module" means to you.
+![](mannish.jpg)
 
-# Usage
+This is a fairly specific implementation of the mediator pattern - given a string name, it calls a function and returns a promise with the function's response.
 
-## Create a new app context
+It accomplishes the same goals as many dependency injection libraries (loose coupling, better testability), but focuses only on functions, and always returning asynchronous responses.
 
-```js
-var mannish = require('mannish')
+Originally inspired by Nicholas Zakas's talk on "[Creating A Scalable JavaScript Application Architecture](http://youtu.be/b5pFv9NB9fs)", this library fills the role of what he called the "sandbox" more closely than it serves the purpose he had for the mediator pattern.
 
-var appContext = mannish()
-```
+Mannish focuses on decoupling the calling of functions and getting a response back.
 
-## Subscribe, adding yourself as a provider
+# API
 
-```js
-appContext.subscribe('unixTimestamp', function(someDate, cb) {
-	cb(null, Math.round(someDate.getTime() / 1000))
-})
-```
+## `mediator = mannish()`
+
+This library exports a single function that returns a new mediator.
 
-Mannish uses promises internally to keep Zalgo at bay and prevent multiple subscribers from answering the same request.
+<!-- js
+const mannish = require('./')
 
-If you just want to subscribe to events and it doesn't make sense to provide a response value, just ignore the callback.
+const mediator = mannish()
+-->
 
-If you want to use promises, then instead of calling the callback you can just return a promise instead.
+## `mediator.provide(name, function)`
 
-## Publish a request for something
+Supplies a function to handle all calls to the given name.
 
 ```js
-appContext.publish('unixTimestamp', new Date(), function(err, result) {
-	// result is a ten-digit number, wheee
+mediator.provide('pie', recipe => {
+	Array.isArray(recipe.contents) // => true
```

Comment:
That `Array.isArray` was actually more just to assert that the code was working the way that it should be when jsmd checks it for correctness.

I do really like the idea of making the example show both both possible return types, though.